### PR TITLE
feat: cases page

### PR DESCRIPTION
**Pull request by LeonidShv**
works only the first page (Cases), for other we need to handle issues with endpoints.
So, you can check only src/pages/Cases.vue
Conclusions, ... and other pages don't work

image

and I have many ideas for improving current solution, so I can add new features step by step

https://github.com/echr-od/ECHR-OD_front-end/assets/57797423/b5ce7737-e7b1-4607-b95f-a9fa30b43304

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -0,0 +1,133 @@
+<template>
+  <header class="header">
+    <VSwitch
+      @update:v-model="onchange"
```
Comment (LeonidShv) [inline src/App.vue]: onChange

File: src/App.vue [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  {
+    name: "Representatives",
+    icon: "",
+    path: "/representatives",
+    label: "Representatives",
+    disabled: true,
+  },
+  {
+    name: "Scl",
+    icon: "",
```
Comment (LeonidShv) [inline src/App.vue]: add and setup icons for nav

File: src/App.vue [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    path: "/scl",
+    label: "SCL",
+    disabled: true,
+  },
+]);
+
+const isDark = ref(JSON.parse(localStorage.getItem('isDark')) || false);
+
+function onchange(value) {
```
Comment (LeonidShv) [inline src/App.vue]: new variables
function updateThemePreference(isDarkModeEnabled) {
  isDark.value = isDarkModeEnabled;
  localStorage.setItem('isDark', isDarkModeEnabled);
}

File: src/components/Navigation/VNavigation.vue [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+import "element-plus/es/components/menu-item/style/css";
+
+defineProps({
+  navigation: {
+    type: Array,
+    default: () => [],
+  },
+  mode: {
+    type: String,
+    default: "",
```
Comment (LeonidShv) [inline src/components/Navigation/VNavigation.vue:37]: add validator, example
validator(value) {
      return ["default", "back"].includes(value);
    },

File: src/components/Pagination/VPagination.vue [hunk 5]
```diff
@@ -0,0 +1,32 @@
+<template>
+  <el-pagination
+    :current-page="currentPage"
+    background
+    layout="prev, pager, next"
+    :total="200"
+    :page-size="20"
+    @update:current-page="test"
```
Comment (LeonidShv) [inline src/components/Pagination/VPagination.vue]: check test

File: src/components/Pagination/VPagination.vue [hunk 6]
```diff
@@ -0,0 +1,32 @@
+<template>
+  <el-pagination
+    :current-page="currentPage"
+    background
+    layout="prev, pager, next"
+    :total="200"
+    :page-size="20"
```
Comment (LeonidShv) [inline src/components/Pagination/VPagination.vue]: check constants

File: src/components/Select/VSelect.vue [hunk 7]
```diff
@@ -0,0 +1,42 @@
+<template>
+  <el-select
+    class="Select"
+    @update:model-value="(value) => $emit('update:v-model', value)"
+    multiple
+    collapse-tags
+    :placeholder="placeholder"
+    style="width: 240px"
```
Comment (LeonidShv) [inline src/components/Select/VSelect.vue]: remove

File: src/components/Navigation/VNavigation.vue [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+  document.querySelector("html").setAttribute("class", className);
+});
+</script>
+
+<style lang="scss" scoped>
+.Navigation {
+  align-items: center;
+  flex-grow: 1;
+
+  &-item {
```
Comment (LeonidShv) [inline src/components/Navigation/VNavigation.vue]: don't use css bliss

File: src/components/Switch/VSwitch.vue [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    default: () => {},
+  },
+  value: {
+    type: Boolean,
+    default: false,
+  },
+});
+</script>
+
+<style lang="scss" scoped></style>
```
Comment (LeonidShv) [inline src/components/Switch/VSwitch.vue]: remove

File: src/components/VError/index.vue [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+import error from "@/assets/animations/error.json";
+</script>
+
+<style lang="scss" scoped>
+.Error {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+
+  &-text {
```
Comment (LeonidShv) [inline src/components/VError/index.vue]: bliss

File: src/helpers/index.js [hunk 11]
```diff
@@ -0,0 +1,13 @@
+export function getReadyTableData(respondedList, configuredList) {
```
Comment (LeonidShv) [inline src/helpers/index.js:1]: check it

File: src/pages/Cases/subPages/CaseIdWrapperPage.vue [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+})
+</script>
+
+<style scoped lang="scss">
+$miv-items-amount: 2;
+.header {
+  justify-content: space-between;
+  min-height: var(--header-height);
+  flex-wrap: wrap;
+  gap: 8px;
```
Comment (LeonidShv) [inline src/pages/Cases/subPages/CaseIdWrapperPage.vue]: gap-8

File: src/pages/Cases/subPages/CaseIdWrapperPage.vue [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+.case-id__navigation {
+  gap: 16px;
```
Comment (LeonidShv) [inline src/pages/Cases/subPages/CaseIdWrapperPage.vue]: gap-16

File: src/pages/NotFound/NotFoundPage.vue [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+import notFoundPage from "@/assets/animations/notFoundPage.json";
+</script>
+
+<style lang="scss" scoped>
+.Error {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+
+  &-text {
```
Comment (LeonidShv) [inline src/pages/NotFound/NotFoundPage.vue]: bliss

File: src/pages/NotFound/NotFoundPage.vue [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+    font-size: 24px;
```
Comment (LeonidShv) [inline src/pages/NotFound/NotFoundPage.vue]: move to constants

File: src/pages/NotFound/NotFoundPage.vue [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
+  &-link {
+    display: block;
+    margin-top: 32px;
```
Comment (LeonidShv) [inline src/pages/NotFound/NotFoundPage.vue]: atomic?

File: server/routes/cases.js [hunk 17]
```diff
@@ -0,0 +1,86 @@
+var express = require('express');
+const axios = require('axios');
+var router = express.Router();
+
+router.get('/', async function(req, res, next) {
+  try {
+    const page = req.query.page || 1;
+    const limit = req.query.limit || 10;
+    const response = await axios.get(`https://echr-opendata.eu/api/v1/cases?page=${page}&limit=${limit}`);
```
Comment (aquemy) [inline server/routes/cases.js]: You might want to define the base url in a global `https://echr-opendata.eu/api/v1/` to avoid repeating it everywhere.

**Review by aquemy — APPROVED** [review]
We use [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) for the whole organization.
Your commits should starts by `feat` or `fix` or `test` or `docs` basically.

It is not an issue here because the PR will be squashed.

File: server/routes/cases.js (hunk 17 above)
Comment (LeonidShv) [reply server/routes/cases.js]: thank you for the feedback

**Pull request merged**